### PR TITLE
Merge Develop into Master - trust policy

### PR DESCRIPTION
**Pull request by codykoelemay**
Adds trustPolicy for createRole and getRole

### CODE REVIEW

File: iam_role.go [hunk 1]
```diff
@@ -114,14 +118,24 @@ func NewIamRoleRequest(options *CreateIamRoleOptions) (*IamRoleRequest, error) {
...
+	if trustPolicyExists == roleTypeExists {
+		return nil, fmt.Errorf("Either RoleType or TrustPolicy must be included, but not both")
 	}
 
 	iam := &IamRoleRequest{
 		RoleName: *options.RoleName,
-		RoleType: *options.RoleType,
 	}
+
+	if options.RoleType != nil {
```
Comment (elliottzack429) [inline iam_role.go]: I would consider using if roleTypeExists variable that you set above.  Not a big deal, but you've already set it and it seems a bit more verbose.

File: iam_role.go [hunk 2]
```diff
@@ -114,14 +118,24 @@ func NewIamRoleRequest(options *CreateIamRoleOptions) (*IamRoleRequest, error) {
...
+		iam.RoleType = *options.RoleType
+	}
+
+	if options.TrustPolicy != nil {
```
Comment (elliottzack429) [inline iam_role.go]: Same Here

File: iam_role.go (hunk 1 above)
Comment (codykoelemay) [reply iam_role.go]: Good Catch

Comment (codykoelemay) [reply iam_role.go]: #48

File: iam_role.go [hunk 3]
```diff
@@ -114,14 +118,24 @@ func NewIamRoleRequest(options *CreateIamRoleOptions) (*IamRoleRequest, error) {
...
 	}
 
 	iam := &IamRoleRequest{
 		RoleName: *options.RoleName,
-		RoleType: *options.RoleType,
 	}
+
+	if roleTypeExists {
+		iam.RoleType = *options.RoleType
+	}
```
Comment (elliottzack429) [inline iam_role.go:133]: Should we be setting this to nil, as we do with other examples such as template fields.

**Review by elliottzack429 — APPROVED** [review]

**Pull request merged**